aten/src/ATen/native/Embedding.cpp

```diff
@@ -19,14 +19,19 @@ Tensor embedding(const Tensor & weight, const Tensor & indices,
 
   // TODO: use tensor.index() after improving perf
   if (indices.dim() == 1) {
-    return weight.index_select(0, indices);
+    auto out = weight.index_select(0, indices);
+    out.masked_fill_((indices == padding_idx).reshape({-1, 1}), 0);
```

Comment:
Out of curiosity, what is the perf implication of this change? The Embedding layer is a fairly perf-critical piece of code AFAIK

Reply:
it is easy to branch conditioned on `padding_idx > 0`. maybe should do that

Reply:
Ah. That makes sense. Thanks @SsnL